code/game/objects/structures/low_wall.dm

```diff
@@ -39,18 +39,32 @@
 	var/hitsound = 'sound/weapons/Genhit.ogg'
 	climbable = TRUE
 
+
+//Low walls mark the turf they're on as a wall.  This is vital for floor icon updating code
+/obj/structure/low_wall/New()
+	var/turf/T = loc
+	if (istype(T))
+		T.is_wall = TRUE
+	.=..()
+
 /obj/structure/low_wall/Destroy()
 	for (var/obj/structure/window/W in loc)
 		if (!QDELETED(W))
 			W.shatter()
 
+	//If we're on a floor, make it no longer be counted as a wall
+	var/turf/simulated/floor/T = loc
+	if (istype(T))
+		T.is_wall = FALSE
```

Comment:
This does not handle a case where there is something else on the turf that set `is_wall` to `TRUE`. To handle such a case, `is_wall` should be a numeric value and each atom that sets `is_wall` to `TRUE` would instead add `1` to it, and on destroy subtract `1` from it.

Reply:
there are no such cases nor are any planned

Reply:
Is there anything preventing two low walls from being on a single turf?

Reply:
Now there is, yes. i pushed a fix to prevent multiple dense objects being crafted on the same tile